app/liquid_tags/youtube_tag.rb

```diff
@@ -26,12 +33,24 @@ def render(_context)
 
   private
 
-  def parse_id(input)
-    input_no_space = input.delete(" ")
-    raise StandardError, "Invalid YouTube ID" unless valid_id?(input_no_space)
-    return translate_start_time(input_no_space) if input_no_space.include?("?t=")
+  def parse_id_or_url(input)
+    if (input =~ YOUTUBE_URL_REGEX)&.zero?
+      extract_youtube_id(input)
+    else
+      input_no_space = input.delete(" ")
+      raise StandardError, "Invalid YouTube ID" unless valid_id?(input_no_space)
+      return translate_start_time(input_no_space) if input_no_space.include?("?t=")
 
-    input_no_space
+      input_no_space
+    end
+  end
+
+  def extract_youtube_id(url)
+    url = url.gsub(/(>|<)/i, "").split(%r{(vi/|v=|/v/|youtu\.be/|/embed/)})
```

Comment:
I don't think this regular expression needs to be case-insensitive.

Reply:
😅  I stole this from an online Gist. I will update accordingly.